### PR TITLE
Set minimum approvals

### PR DESCRIPTION
**Pull request by elmundio87**
This PR allows setting a minimum number of approvals before the workflow continues, via a new input parameter `minimum_approvals`.

This means you can now list an entire team of users as `approvers`, but only need one of them to approve the manual step.

If the new parameter is not set, the existing behaviour is preserved - all reviewers are required to approve.

I have updated the test suite to take this change into account, and I compiled my own docker image to test in a separate repository.

Passing workflow: https://github.com/elmundio87/manual-approval-test/runs/5706511606?check_suite_focus=true
Associated issue: https://github.com/elmundio87/manual-approval-test/issues/5

Thanks for the great Github Action!

### CODE REVIEW

File: action.yaml
Comment (trstringer) [inline action.yaml:10]: Is this format (underscore separating words) the accepted and common standard for actions inputs? Can you link an example of another action that uses this format? Not saying it's wrong, just want to make sure that we're following any standard that might exist.

File: approval_test.go
Comment (trstringer) [inline approval_test.go:148]: Can you add a test case with multiple approvers and the number of comment approvals is less than `minimumApprovals`? I'd like to make sure we cover the case where we have *some* approvals, but not enough to meet minimum approvals. So perhaps 3 approvers, `minimumApprovals` set to 2, and only 1 approved message.

File: main.go
Comment (trstringer) [inline main.go:51]: I agree with the exiting with a non-zero status code but this message shouldn't be "warning", it should be "error".

```suggestion
		fmt.Printf("error: minimum required approvals (%v) is greater than the total number of approvers (%v)\n", minimumApprovals, len(approvers))
```

**Review by trstringer — CHANGES_REQUESTED** [review]
Thanks for the contribution! Just a couple of small changes before merging.

File: action.yaml
Comment (elmundio87) [reply action.yaml:10]: This is a fair comment - in actions created by Github, they prefer hyphen separated inputs as a standard. Happy to change this.

File: main.go
Comment (elmundio87) [reply main.go]: Agreed, I have updated this to an error

File: approval_test.go
Comment (elmundio87) [reply approval_test.go:148]: I have added the test case `multi_approver_approvals_less_than_minimum` to cover this

**Comment by elmundio87** [comment]
Hi @trstringer , I have made the changes you have requested

**Comment by gerardstanczak-leeroy** [comment]
Love the feature, please merge it to make it available for general use :)

**Comment by gerardstanczak** [comment]
Love the feature, please merge it to make it available for general use :)

**Comment by golgotow** [comment]
this be great, waiting for merge !

**Comment by trstringer** [comment]
Thank you for your contribution! 🎉

**Pull request merged**
